### DISCUSSION TITLE
关于 UI 组件绑定代码的问题

### DISCUSSION THREAD

**Discussion by iiitong** [jynew/jynew#1071] [Q&A] [answered]
文档上说“代码绑定的方式会方便查找代码引用”，不是很理解这句话的意思，通过拖拽的方式不是也可以方便在代码里查找所有引用吗？

本人对 Unity 不是很熟，网上搜到的对比都是说用 `transform.find` 可以解决 meta 丢失时候需要重复拖拽的问题。有没有大佬可以对比一下这两种方案的区别啊？十分感谢🙏

**Comment by 0kk470** [comment, marked as answer]
两种方式都用过，以下仅为个人意见。

### 代码获取路径绑定

优点

* 可以写一个模板工具根据一定的规则来生成绑定代码，后期维护起来比较方便。
* 用代码绑定某种意义上来也就会区分哪些控件是废弃掉的，哪些是还在使用的，而且也能通过版本管理软件快速查看改动的部分，便于熟悉业务逻辑。

缺点

* 初始化时会有Find路径以及获取控件的开销， 代码量有所上升。
* 控件较多的情况下，如果手写容易打错控件路径。

### 拖拽方式

优点
*  可视化拖拽编辑，便于操作，可以快速出效果。

缺点
*  会有丢失引用的风险， 协作的人越多这个问题出现的频率越高。
*  **SerializeField** 会生成额外的序列化字段，prefab文件需要存储的数据会变多，占用更多的磁盘和内存空间。
           
### 总结

个人觉得拖拽的方式适合单人或者维护成本较低的中小型项目，代码绑定这种适合需要多人协作、有一定维护标准要求的大型项目。

**Reply by iiitong** [reply]
非常感谢！
